Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,108 @@
+﻿using UnityEngine.Audio;
+using UnityEngine;
+using System;
+using System.Collections;
+
+public class AudioManager : MonoBehaviour
+{
+    public AudioMixer mainMixer;
+    public Sound[] sounds;
+    public Sound[] triggerIndependentSounds;
+    public float minTimeBetweenAmbient;
+    public float maxTimeBetweenAmbient;
+
+    Sound currentAmbient;
+
+    private void Awake()
+    {
+        App.audioManager = this;
+
+        foreach (Sound s in sounds)
+        {
+            s.source = gameObject.AddComponent<AudioSource>();
+            s.source.clip = s.clip;
+            s.source.outputAudioMixerGroup = s.output;
+            s.source.volume = s.volume;
+            s.source.loop = s.loop;
+        }
+
+        foreach (Sound s in triggerIndependentSounds)
+        {
+            s.source = gameObject.AddComponent<AudioSource>();
+            s.source.clip = s.clip;
+            s.source.outputAudioMixerGroup = s.output;
+            s.source.volume = s.volume;
+            s.source.loop = s.loop;
+        }
+    }
+
+    private void Start()
+    {
+        LoadVolumeValues();
+    }
+
+    public void Play(string name)
+    {
+        Sound s = FindSound(name);
```

Comment:
you might want to check if not null

Reply:
to sa checkne v FindSound() funkcii a hodí LogError keď taký zvuk neni